src/daft-logical-plan/src/builder/mod.rs

```diff
@@ -243,6 +243,26 @@ impl LogicalPlanBuilder {
         Ok(self.with_new_plan(logical_plan))
     }
 
+    pub fn with_columns_renamed(&self, cols_map: HashMap<String, String>) -> DaftResult<Self> {
+        let exprs = self
+            .schema()
+            .fields
+            .iter()
+            .map(|(name, _)| {
+                if let Some(new_name) = cols_map.get(name) {
+                    // If the column is in the rename map, create an alias expression
+                    col(name.clone()).alias(new_name.clone())
+                } else {
+                    // Otherwise keep the original column reference
+                    col(name.clone())
```

Comment:
```suggestion
                    col(name.as_str()).alias(new_name.as_str())
                } else {
                    // Otherwise keep the original column reference
                    col(name.as_str())
```

Nit: Use `as_str` instead of clone to elide heap allocation